homeassistant/setup.py

```diff
@@ -685,10 +689,31 @@ def async_pause_setup(
     try:
         yield
     finally:
-        time_taken = time.monotonic() - started
         integration, group = running
-        # Add negative time for the time we waited
-        _setup_times(hass)[integration][group][phase] = -time_taken
+        if running not in _setup_started(hass):
```

Comment:
I think we can actually move this check up at the start, it doesn't need to wait

Reply:
Yup, and thats much much better because it keeps things a lot simplier